scalgrafanalib/core.py

```diff
@@ -99,19 +131,96 @@ def to_json_data(self):
         }
 
 
+@attr.s
+class StateMapping:
+    """
+    Represents a state mapping for StateTimeline panels with color and text
+    """
+
+    value: Union[int, str, None] = attr.ib()
+    text: str = attr.ib()
+    color: Optional[str] = attr.ib(default=None)
+    index: int = attr.ib(default=0)
+
+    def to_json_data(self) -> Dict[str, Any]:
+        result = {
+            "index": self.index,
+            "text": self.text,
+        }
+        if self.color:
+            result["color"] = self.color
+        return result
+
+
 @attr.s
 class StateTimeline(core.StateTimeline):
-    """StateTimeline: Allow settings minValue and maxValue"""
+    """StateTimeline: Allow settings minValue, maxValue, and color mappings for MongoDB states"""
 
     minValue = attr.ib(default=None)  # pylint: disable=invalid-name
     maxValue = attr.ib(default=None)  # pylint: disable=invalid-name
+    colorMode: str = attr.ib(default="palette-classic")  # pylint: disable=invalid-name
+    mappings: List[Union[StateMapping, Dict[str, Any]]] = attr.ib(
+        default=None
+    )  # pylint: disable=invalid-name
 
     def to_json_data(self) -> Json:
         json = super().to_json_data()
         if self.minValue:
             json["fieldConfig"]["defaults"]["min"] = self.minValue
         if self.maxValue:
             json["fieldConfig"]["defaults"]["max"] = self.maxValue
+
+        # Set color mode
+        if "fieldConfig" not in json:
+            json["fieldConfig"] = {}
+        if "defaults" not in json["fieldConfig"]:
+            json["fieldConfig"]["defaults"] = {}
+        if "color" not in json["fieldConfig"]["defaults"]:
+            json["fieldConfig"]["defaults"]["color"] = {}
```

Comment:
checks not needed: garanteed by grafanalib implementation (c.f. earlier use in lin 169-171)